shared/fs/index.js

```diff
@@ -2,156 +2,89 @@
 import * as I from 'immutable'
 import * as React from 'react'
 import * as Types from '../constants/types/fs'
-import {globalStyles, globalMargins} from '../styles'
-import {Box, Divider, Icon, List, ScrollView, Text} from '../common-adapters'
+import * as Styles from '../styles'
+import * as Kb from '../common-adapters'
 import FolderHeader from './header/container'
 import SortBar from './sortbar/container'
-import {TlfType, Tlf, Still, Editing, Placeholder, Uploading} from './row/containers'
 import Footer from './footer/footer'
 import {isMobile} from '../constants/platform'
 import ConnectedResetBanner from './banner/reset-banner/container'
+import Rows from './row/rows-container'
 
 type FolderProps = {
   isUserReset: boolean,
-  items: Array<Types.RowItem>,
+  sortSetting: Types.SortSetting,
   path: Types.Path,
   resetParticipants: Array<string>,
   routePath: I.List<string>,
 }
 
-export const WrapRow = ({children}: {children: React.Node}) => (
-  <Box style={stylesRowContainer}>
-    {children}
-    <Divider key="divider" style={stylesDivider} />
-  </Box>
-)
-
 class Files extends React.PureComponent<FolderProps> {
-  _rowRenderer = (index: number, item: Types.RowItem) => {
-    switch (item.rowType) {
-      case 'placeholder':
-        return (
-          <WrapRow key={`placeholder:${item.name}`}>
-            <Placeholder type={item.type} />
-          </WrapRow>
-        )
-      case 'tlf-type':
-        return (
-          <WrapRow key={`still:${item.name}`}>
-            <TlfType name={item.name} routePath={this.props.routePath} />
-          </WrapRow>
-        )
-      case 'tlf':
-        return (
-          <WrapRow key={`still:${item.name}`}>
-            <Tlf name={item.name} tlfType={item.tlfType} routePath={this.props.routePath} />
-          </WrapRow>
-        )
-      case 'still':
-        return (
-          <WrapRow key={`still:${item.name}`}>
-            <Still name={item.name} path={item.path} routePath={this.props.routePath} />
-          </WrapRow>
-        )
-      case 'uploading':
-        return (
-          <WrapRow key={`uploading:${item.name}`}>
-            <Uploading name={item.name} path={item.path} />
-          </WrapRow>
-        )
-      case 'editing':
-        return (
-          <WrapRow key={`editing:${Types.editIDToString(item.editID)}`}>
-            <Editing editID={item.editID} routePath={this.props.routePath} />
-          </WrapRow>
-        )
-      default:
-        /*::
-      let rowType = item.rowType
-      declare var ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove: (rowType: empty) => any
-      ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove(rowType);
-      */
-        return (
-          <WrapRow key="">
-            <Text type="BodySmallError">This should not happen.</Text>
-          </WrapRow>
-        )
-    }
-  }
-
   render() {
+    console.log({songgao: 'Files', path: this.props.path, props: this.props})
```

Comment:
🔥

Reply:
This is the fire that warmed up the code freeze 😎